### ISSUE TITLE
[Bug] Searching Crash list by crash ID not working as expected when field is not set

### ISSUE THREAD

**Issue by mddilley** [cityofaustin/atd-data-tech#15454]
When searching the Crash list page by crash id, the view will initially allow you to search without setting a field in the dropdown. The UI shows that the list is being filtered by the ID that you enter, but the list results are not filtered to that ID. This can be confirmed by looking at the query request payload.

**Update: This only happens the very first time a user searches ever. Once the local storage config entry is saved like `jestersavedCrashesConfig`, the bug will never reappear.**

To reproduce:
1. Go to the Crash list page
2. Find a crash ID in the list and paste it into the search box
3. Press **Enter** and see that the UI show a badge for the crash ID but the results are not filtered to that ID
4. Also, there is usually a yellow alert that shows when a field has not been set in the dropdown to signal that the search filter isn't going to apply
5. This only happens after you set a field in the dropdown for the first time

**Screen capture:**
<img width="1075" alt="Screenshot 2024-01-29 at 6 01 16 PM" src="https://github.com/cityofaustin/atd-data-tech/assets/37249039/a3c00107-3a16-4d5e-bb30-de6d350f22f1">

**Comment by mddilley** [comment]
@patrickm02L This is another small one that VZ users have probably worked around.

**Comment by patrickm02L** [comment]
Thanks @mddilley, I think I've experienced this myself. 

<img width="1454" alt="Screen Shot 2024-01-29 at 8 43 38 PM" src="https://github.com/cityofaustin/atd-data-tech/assets/5566727/83bd9042-7ff4-4c55-8c52-3d6fae8c1334">

**Comment by mddilley** [comment]
@johnclary I ran into some code that we shipped in a patch to fix a bug around incompatible GridTable local storage settings, but [you and Frank had great ideas on more long term fix](https://github.com/cityofaustin/atd-vz-data/pull/1207#pullrequestreview-1365842135). Might be another good snackoo and also there is this [gnarly useEffect in GridTable](https://github.com/cityofaustin/atd-vz-data/blob/30e42ecdb6eab46dc9c1544d9eccaba0ff0bf42a/atd-vze/src/Components/GridTable.js#L97-L114) from long ago that I also tripped over and didn't want to just pass over.

I'll reach out tomorrow to see how you feel about broadening the scope of this issue tomorrow, or I can also write it up in another one.
